src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -939,29 +948,26 @@ export class StudyViewPageStore
             }
 
             // group present in page session which are not saved to user account
-            const missingGroupIds = Array.from(
-                this._selectedComparisonGroups.keys()
+            const missingGroupIds: string[] = Array.from(
+                this.createdGroups.keys()
             ).filter(groupId => groupIdSet[groupId] === undefined);
 
             if (missingGroupIds.length > 0) {
-                const promises = [];
-                for (const groupId of missingGroupIds) {
-                    promises.push(comparisonClient.getGroup(groupId));
-                }
+                const promises = missingGroupIds.map(groupId =>
+                    comparisonClient.getGroup(groupId)
+                );
                 const studyIdsSet = stringListToSet(
                     this.queriedPhysicalStudyIds.result!
                 );
                 let newGroups: Group[] = await Promise.all(promises);
 
-                newGroups
-                    .filter(
-                        group =>
-                            !_.some(
-                                group.data.studies,
-                                study => studyIdsSet[study.id] === undefined
-                            )
-                    )
-                    .forEach(group =>
+                newGroups.forEach(group => {
+                    if (
+                        !_.some(
```

Comment:
this would be more clear written:
```
_.every(studies, (s)=>s.id in studyIdsSet)
```
Also can you put a comment explaining why we would might sometimes get a group with studies that are not in the current queriedPhysicalStudies?